lib/galaxy/tool_util/linters/output.py

```diff
@@ -76,6 +77,23 @@ def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
             names.add(name)
 
 
+class OutputsFilterExpression(Linter):
+    @classmethod
+    def lint(cls, tool_source: "ToolSource", lint_ctx: "LintContext"):
+        tool_xml = getattr(tool_source, "xml_tree", None)
+        if not tool_xml:
+            return
+        for filter in tool_xml.findall("./outputs//filter"):
+            try:
+                ast.parse(filter.text, mode="eval")
+            except Exception as e:
+                lint_ctx.error(
```

Comment:
To me a linter error was always a wrong behavior (e.g. a tool that always produces an output if there should be some cases where it should not). But a definition of error/warning is missing. But, in practice the distinction seems obsolete anyway (IMO) since tool linting CI fails on both.